vms/rpcchainvm/vm_client.go

```diff
@@ -866,6 +866,13 @@ func (b *blockClient) Parent() ids.ID {
 	return b.parentID
 }
 
+func (b *blockClient) VerifyProposer(ctx context.Context) error {
+	_, err := b.vm.client.BlockVerifyProposer(ctx, &vmpb.BlockVerifyProposerRequest{
```

Comment:
Why is this required if the VM always returns `nil`? https://github.com/ava-labs/coreth/commit/248684574146f5425889b9b6cd47226f245cbba3

Is this an optimistic future where the VM would manage proposer validity?